# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/21 08:03:15-08:00 jdike@addtoit.com 
#   [PATCH] uml: fix a stack corruption crash
#   
#   Fix a race where signals could be handled to the parent of a new process on
#   the kernel stack of the child, corrupting that stack, and crashing UML when
#   the next first runs.
#   
#   Signed-off-by: Jeff Dike <jdike@addtoit.com>
#   Signed-off-by: Andrew Morton <akpm@osdl.org>
#   Signed-off-by: Linus Torvalds <torvalds@osdl.org>
# 
# arch/um/kernel/skas/process.c
#   2005/01/20 20:59:16-08:00 jdike@addtoit.com +2 -1
#   uml: fix a stack corruption crash
# 
diff -Nru a/arch/um/kernel/skas/process.c b/arch/um/kernel/skas/process.c
--- a/arch/um/kernel/skas/process.c	2005-01-22 01:49:12 -08:00
+++ b/arch/um/kernel/skas/process.c	2005-01-22 01:49:12 -08:00
@@ -224,9 +224,10 @@
 	block_signals();
 	if(sigsetjmp(fork_buf, 1) == 0)
 		new_thread_proc(stack, handler);
-	set_signals(flags);
 
 	remove_sigstack();
+
+	set_signals(flags);
 }
 
 void thread_wait(void *sw, void *fb)
